internal/api/connectors/sources/impl.go

```diff
@@ -0,0 +1,146 @@
+package sources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"playbook-dispatcher/internal/common/constants"
+	"playbook-dispatcher/internal/common/utils"
+	"time"
+
+	"github.com/redhatinsights/platform-go-middlewares/request_id"
+	"github.com/spf13/viper"
+)
+
+const (
+	basePath               = "/api/sources/v3.1/"
+	filterPath QueryFilter = "filter[source_ref][eq]="
+)
+
+type SourcesClient interface {
+	GetSources(
+		ctx context.Context,
+		sourceId ID,
+	) (sources []*GetSourcesResponse, err error)
+
+	GetRHCConnectionStatus(
+		ctx context.Context,
+		sourceId ID,
+	) (status *GetRhcConnectionStatusResponse, err error)
+}
+
+type sourcesClientImpl struct {
+	client ClientWithResponsesInterface
+}
+
+func createGetSourcesResponseList(resp *[]Source) []*GetSourcesResponse {
+	sourceResponseList := make([]*GetSourcesResponse, len(*resp))
+	for i, response := range *resp {
+		sourceResponseList[i] = &GetSourcesResponse{
+			ID:                 string(*response.Id),
+			Name:               *response.Name,
+			AvailabilityStatus: *response.AvailabilityStatus,
+		}
+	}
+
+	return sourceResponseList
+}
+
+func createGetRhcConnectionStatus(resp *RhcConnectionRead) *GetRhcConnectionStatusResponse {
```

Comment:
Like above...
I would remove the pointers where necessary.

If you do have to pass in a pointer into a function, then I would check for nil in that function.  Have the function protect itself!  :)